### ISSUE TITLE
Make a Lua DrawTask object

### ISSUE THREAD

**Issue by lhog** [beyond-all-reason/spring#685]
Such that things like `glBallisticCircle()` and `glSurfaceCircle()` results can be exposed as objects to Lua space where they could be used to store and draw cached stuff.

**Comment by lhog** [comment]
Also see https://github.com/beyond-all-reason/spring/issues/679

**Comment by Beherith** [comment]
I would recommend these also be instancevbotable based, there are fine examples of this in the sensor ranges widgets.

**Comment by lhog** [comment]
It's not only about VBO/VAO, but also about needed FFP state, shaders, shader uniforms and draw type (array,index,instance) and draw primitives. The new Lua object should encapsulate all that.

Also we don't want Lua to do stuff `glBallisticCircle` does for example as it involves a ton of computations, including some of them done in MT sections.

**Comment by Beherith** [comment]
Actually, doing the ballistics in vertex shader is quite ok, for cannon projectiles, i did that for an experimental widget

**Comment by lhog** [comment]
Yes, but it's an advanced usage and `DrawTask` can be reused across many applications as a replacement for display lists (though each such application will need in-engine implementation to produce a `DrawTask`).

**Comment by Krogoth100** [comment]
> Yes, but it's an advanced usage and `DrawTask` can be reused across many applications as a replacement for display lists (though each such application will need in-engine implementation to produce a `DrawTask`).

Can you give more info on how DrawTask is supposed to work from the Lua side.
Let's say, each frame I determine N objects and render them with some constant BlendType, some constant DepthTestType and some constant CullingType.
What the Lua code would be.

**Comment by lhog** [comment]
I see it more like a returnable result from certain engine calls that carries enough information to (re)draw the same thing once again.

**Comment by Krogoth100** [comment]
> 

Gotcha, its all about static data.
This is what is often called "command buffer".
Gotta make research before doing some input, but from the first glance it's not so easy, like what the best practices are and what not (outside trivial usage).
